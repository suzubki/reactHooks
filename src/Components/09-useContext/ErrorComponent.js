import React from 'react'

export const ErrorComponent = () => {
  return (
    <div>ErrorComponent</div>
  )
}
