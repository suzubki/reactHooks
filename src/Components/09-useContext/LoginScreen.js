import React, { useContext } from 'react'
import { UserContext } from './userContext'

export const LoginScreen = () => {

  const { setUser } = useContext( UserContext )

  const newUser = {
    id:'123',
    name: 'darwin'
  }

  return (
    <div>
        <h1>LoginScreen </h1>
        <hr/>

        <button
          className='btn btn-primary'
          onClick={ () => setUser({...newUser})}
        >
          Establecer nuevo valor
        </button>

    </div>
  )
}
