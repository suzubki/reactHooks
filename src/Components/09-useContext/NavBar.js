import React from 'react'
import { NavLink } from "react-router-dom";

export const NavBar = () => {
  return (
    <nav className="navbar navbar-expand-lg navbar-dark bg-dark">
      <div className="container-fluid">
        <div className="collapse navbar-collapse" id="navbarNav">
          <ul className="navbar-nav">
            <li className='nav-item'>
              <NavLink className='nav-link text-white' aria-current="page" to='/'>HomeScreen</NavLink>
            </li>
            <li className='nav-item'>
              <NavLink className='nav-link text-white' to='/about'>AboutScreen</NavLink>
            </li>
            <li className='nav-item'>
              <NavLink className='nav-link text-white' to='/login'>LoginScreen</NavLink>
            </li>
          </ul>
        </div>
      </div>
    </nav>

  )
}
