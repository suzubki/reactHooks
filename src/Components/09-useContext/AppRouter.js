import React from 'react'
import {
    BrowserRouter,
    Routes,
    Route,
} from "react-router-dom";
import { AboutScreen } from './AboutScreen';
import { LoginScreen } from './LoginScreen';
import { HomeScreen } from './HomeScreen';
import { ErrorComponent } from './ErrorComponent';
import { NavBar } from './NavBar';

export const AppRouter = () => {
    return (
        <div>
        <BrowserRouter>
            <NavBar />
            <Routes>    
                <Route path="/" element={<HomeScreen />} />

                <Route path="/about" element={<AboutScreen />} />
                <Route path="/login" element={<LoginScreen />} />
                <Route path='*' element={<ErrorComponent />} />
            </Routes>
        </BrowserRouter>
        </div>
  )
}
