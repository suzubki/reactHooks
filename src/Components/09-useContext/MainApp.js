import React, { useState } from 'react'
import { AppRouter } from './AppRouter'
import { UserContext } from './userContext'

export const MainApp = () => {

    const [user, setUser] = useState({})

    return (
        <div>
            <UserContext.Provider value={ {
                user,
                setUser
            } } >
                <AppRouter />
            </UserContext.Provider>
        </div>
    )
}
