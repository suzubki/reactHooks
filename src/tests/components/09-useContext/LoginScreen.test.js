import { mount } from "enzyme";
import { LoginScreen } from "../../../Components/09-useContext/LoginScreen";
import { UserContext } from "../../../Components/09-useContext/userContext";

describe("<LoginScreen />", () => {
    const setUser = jest.fn();

    const wrapper = mount(
        <UserContext.Provider value={{ setUser }}>
            <LoginScreen />
        </UserContext.Provider>
    );

    test("should display", () => {
        expect(wrapper).toMatchSnapshot();
    });

    const newUser = {
        id: "123",
        name: "darwin",
    };

    test("should pass newUser", () => {
        wrapper.find("button").simulate("click");
        expect(setUser).toHaveBeenCalledWith(newUser);
    });
});
