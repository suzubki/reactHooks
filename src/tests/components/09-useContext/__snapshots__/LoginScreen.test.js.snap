// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginScreen /> should display 1`] = `
<div>
  <h1>
    LoginScreen 
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Establecer nuevo valor
  </button>
</div>
`;
